tests/unit-tests/conversions.cpp

```diff
@@ -59,4 +51,107 @@ TEST_CASE("C++ Standard Conversions") {
       lexicon.double_type()
   );
   lexicon.make_if_then(condition, *lexicon.make_expr_stmt(then_expr));
+
+  // Bitcast                      (Pretend)
+  INFO("(bool*)ptr");
+  lexicon.make_pretend(
+    *lexicon.make_id_expr(ptr),
+    lexicon.get_pointer(lexicon.bool_type()),
+    lexicon.get_pointer(lexicon.bool_type())
+  );
+}
+
+TEST_CASE("Class Conversions") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  INFO("struct Base {};");
+  INFO("struct Derived : Base {};");
+  auto& base = *lexicon.make_class(*unit.global_region());
+  auto& derived = *lexicon.make_class(*unit.global_region());
+  derived.declare_base(base);
+
+  INFO("Base b;");
+  INFO("Derived d;");
+  auto& base_ptr = lexicon.get_pointer(base);
+  auto& derived_ptr = lexicon.get_pointer(derived);
+  auto& b = *unit.global_region()->declare_var(*lexicon.make_identifier("b"), base_ptr);
+  auto& d = *unit.global_region()->declare_var(*lexicon.make_identifier("d"), derived_ptr);
+
+  // Derived-to-base conversion
+  INFO("(Base*)b;");
+  lexicon.make_narrow(
+    *lexicon.make_id_expr(d),
+    base,
+    base_ptr
+  );
+
+  // Checked base-to-derived conversion
+  INFO("dynamic_cast<Derived*>(b);");
+  lexicon.make_widen(
```

Comment:
Are these mixed up?  I expect 
  - `Narrow` to be a base-to-derived conversion (which generally maps to `dynamic_cast` in a single inheritance hierarchy, but `dynamic_cast` is much more than that)
  - `Widen` to be a derived-to-base conversion

Reply:
This was a typo, I did mean to have the reverse.

Reply:
This test case is to illustrate how to construct the different nodes in the IPR using practical examples, focusing on the nodes not the C++ language.  This is not a guide on how to write a compiler so explaining what Dynamic_cast is and how it might be represented differently across different uses is not really in scope.